src/Button/MeasureButton/MeasureButton.jsx

```diff
@@ -0,0 +1,747 @@
+    drawInteraction.on('change:active', this.onDrawInteractionActiveChange, this);
+    map.addInteraction(drawInteraction);
+
+    this.setState({drawInteraction});
+  }
+
+  /**
+   * Adjusts visibility of measurement related tooltips depending on active
+   * status of draw interaction.
+   *
+   * @method
+   */
+  onDrawInteractionActiveChange = () => {
+    if (this.state.drawInteraction.getActive()) {
+      this.createHelpTooltip();
+      this.createMeasureTooltip();
+    } else {
+      this.removeHelpTooltip();
+      this.removeMeasureTooltip();
+    }
+  }
+
+  /**
+   * Sets up listeners whenever the drawing of a measurement sketch is
+   * started.
+   *
+   * @param {ol.interaction.Draw.Event} evt The event which contains the
+   *   feature we are drawing.
+   *
+   * @method
+   */
+  drawStart = (evt) => {
+    const {
+      showMeasureInfoOnClickedPoints,
+      multipleDrawing,
+      measureType,
+      map
+    } = this.props;
+
+    const source = this.state.measureLayer.getSource();
+    this._feature = evt.feature;
+
+    if (showMeasureInfoOnClickedPoints && measureType === 'line') {
+      this._eventKeys.click = map.on('click', this.addMeasureStopTooltip, this);
+    }
+
+    if (!multipleDrawing && source.getFeatures().length > 0) {
+      this.cleanupTooltips();
+      this.createMeasureTooltip();
+      this.createHelpTooltip();
+      source.clear();
+    }
+  }
+
+  /**
+   * Called whenever measuring stops, this method draws static tooltips with
+   * the result onto the map canvas and unregisters various listeners.
+   *
+   * @method
+   */
+  drawEnd = () => {
+    const {
+      measureType,
+      showMeasureInfoOnClickedPoints,
+      measureTooltipCssClasses
+    } = this.props;
+
+    if (this._eventKeys.click) {
+      OlObservable.unByKey(this._eventKeys.click);
+    }
+
+    // Fix doubled label for lastPoint of line
+    if (showMeasureInfoOnClickedPoints && measureType === 'line') {
+      this.removeMeasureTooltip();
+    } else {
+      this._measureTooltipElement.className =
+        `${measureTooltipCssClasses.tooltip} ${measureTooltipCssClasses.tooltipStatic}`;
+      this._measureTooltip.setOffset([0, -7]);
+    }
+
+    // unset sketch
+    this._feature = null;
+
+    // fix doubled label for last point of line
+    if (showMeasureInfoOnClickedPoints && measureType === 'line') {
+      this._measureTooltipElement = null;
+      this.createMeasureTooltip();
+    }
+  }
+
+  /**
+   * Adds a tooltip on click where a measuring stop occured.
+   *
+   * @param {ol.MapBrowserEvent} evt The event which contains the coordinate
+   *   for the tooltip.
+   *
+   * @method
+   */
+  addMeasureStopTooltip = (evt) => {
+    const {
+      measureType,
+      decimalPlacesInTooltips,
+      map,
+      measureTooltipCssClasses
+    } = this.props;
+
+    if (!isEmpty(this._feature)) {
+      const geom = this._feature.getGeometry();
+      let value = measureType === 'line' ?
```

Comment:
You may use `const` here.

Reply:
Replaced here and somewhere else via 216e403